### DISCUSSION TITLE
Compare values and update only when different from previous

### DISCUSSION THREAD

**Discussion by lubianat** [LeMyst/WikibaseIntegrator#597] [Q&A] [answered]
Hi! 
I have some code that pull some items from my wikibase, adds (or not) information, and then tries to write to the Wikibase: 
This is the final snippet:

```
            item.claims.add(data, action_if_exists=ActionIfExists.REPLACE_ALL)

            old_item_dict = old_item.__dict__
            new_item_dict = item.__dict__

            if old_item_dict == new_item_dict:
                print("No change")
            else:
                try:
                    item.write()
                except ModificationFailed as e:
                    print(e)
                    pass
 ```
 
 This comparison, however, is not working. I want to implement that ifelse statement to avoid talking to the server whenever I don't have anything new to write. Any thoughts? 
 
 Thanks!

**Comment by LeMyst** [comment, marked as answer]
Hello @lubianat ,

I can't test right now, but when you say the comparison didn't work, it's always true or false?
You didn't precise how you create the old_item and new_item, but if you did simply a copy, you may need to you the deepcopy function from the copy package.
Otherwise you can try to compare the JSON object from each item (but you need a deepcopy too)

I hope that can help you,

**Comment by lubianat** [comment]
@LeMyst it is always true; I create the item by 

```
            item = wbi.item.get(entity_id=proposal_qid)
```


maybe the json is the way

**Comment by LeMyst** [comment]
You can try the
`old_item = copy.deepcopy(item)`

**Comment by lubianat** [comment]
the json trick worked, thanks! This one should work too.
